### PR TITLE
docs: github docs

### PR DESCRIPTION
**Pull request by tedim52**
## Description:
Adds a guide on how to run private packages with `kurtosis github login`.

Other changes:
- Adds docs for the github CLI commands. 
- Restarts the engine after successful login so user doesn't have to remember to do it
- Makes OAuth perms more restrictive 
- Adds log retention info to service logs bc it was missing

## Is this change user facing?
YES

## References:
https://github.com/kurtosis-tech/kurtosis/issues/2183
https://github.com/kurtosis-tech/kurtosis/issues/2181

### CODE REVIEW

File: cli/cli/helpers/oauth/oauth.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ type OAuth interface {
 func AuthFlow() (string, string, error) {
 	httpClient := &http.Client{} // nolint: exhaustruct
 
-	minimumScopes := []string{"repo", "read:org", "gist"}
+	minimumScopes := []string{"repo"}
```
Comment (h4ck3rk3y) [inline cli/cli/helpers/oauth/oauth.go:43]: can this be read:repo? defaults to asking for write and read

File: cli/cli/helpers/oauth/oauth.go [hunk 2]
```diff
@@ -51,6 +51,7 @@ func AuthFlow() (string, string, error) {
 		Scopes:       minimumScopes,
 		DisplayCode: func(code, verificationURL string) error {
 			fmt.Fprintf(os.Stdout, "First copy your one-time code: %s\n", code)
+			fmt.Fprintf(os.Stdout, "Then, press Enter to be directed to a browser window where you'll enter this code.\n")
```
Comment (h4ck3rk3y) [inline cli/cli/helpers/oauth/oauth.go:54]: Can it be auto entered?

File: cli/cli/commands/github/login/login.go [hunk 3]
```diff
@@ -46,5 +54,30 @@ func run(_ context.Context, _ *flags.ParsedFlags, _ *args.ParsedArgs) error {
...
+}
+
+func RestartEngineAfterGitHubAuth(ctx context.Context) error {
+	engineManager, err := engine_manager.NewEngineManager(ctx)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred creating an engine manager.")
+	}
+	var engineClientCloseFunc func() error
+	var restartEngineErr error
+	_, engineClientCloseFunc, restartEngineErr = engineManager.RestartEngineIdempotently(ctx, defaults.DefaultEngineLogLevel, defaultEngineVersion, restartEngineOnSameVersionIfAnyRunning, defaults.DefaultEngineEnclavePoolSize, defaults.DefaultEnableDebugMode, defaults.DefaultGitHubAuthTokenOverride)
```
Comment (h4ck3rk3y) [inline cli/cli/commands/github/login/login.go:73]: we lose some of these other configurations; guess thats okay

File: docs/docs/cli-reference/github-logout.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+---
+title: github logout
+sidebar_label: github logout
+slug: /github-logout
+---
+
+Logs out a GitHub user by removing the OAuth token for authorizing GitHub operations from secure system credential storage or plain text file.
```
Comment (h4ck3rk3y) [inline docs/docs/cli-reference/github-logout.md]: which one is it plain text file or secure credential storage?

File: docs/docs/cli-reference/service-logs.md [hunk 5]
```diff
@@ -6,14 +6,18 @@ slug: /service-logs
...
 kurtosis service logs $THE_ENCLAVE_IDENTIFIER $THE_SERVICE_IDENTIFIER
 ```
-
 where `$THE_ENCLAVE_IDENTIFIER` and the `$THE_SERVICE_IDENTIFIER` are [resource identifiers](../advanced-concepts/resource-identifier.md) for the enclave and service, respectively. The service identifier (name or UUID) is printed upon inspecting an enclave. 
+:::
 
 :::note Number of log lines
 By default, logs printed in the terminal from this command are truncated at the most recent 200 log lines. For a stream of logs, we recommend the `-f` flag. For all the logs use the `-a` flag and for a snapshot of the logs at a given point in time (e.g. after a change), we recommend the [`kurtosis dump`](./dump.md).
+
+:::note Log Retention
```
Comment (h4ck3rk3y) [inline docs/docs/cli-reference/service-logs.md:19]: this should have been a separate PR!

File: docs/docs/guides/running-private-packages.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+The following prompt should be displayed. After copying the one time code, press enter.
+
+![github-one-time-code](../../static/img/guides/one-time-code.jpg)
+
+A GitHub screen should pop up in your browser instructing you to enter the one-time code you just copied in your terminal. 
+
+![github-enter-code](../../static/img/guides/github-enter-code.jpg)
+
+After entering the code, GitHub will prompt you to authorize Kurtosis CLI. Kurtosis CLI only requests read access to public and private repositories.
```
Comment (h4ck3rk3y) [inline docs/docs/guides/running-private-packages.md]: for now its both read and write until we scope the token

File: docs/docs/guides/running-private-packages.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+
+:::info OAuth Token storage
+Behind the scenes, authorizing Kurtosis CLI means GitHub will generate an OAuth token that Kurtosis CLI can use to perform GitHub operations on your behalf. Following GitHub CLI, we attempt to securely store this token in system credential storage. If system credential store is unavailable, we store it in a plain text file in your Kurtosis config directory at `kurtosis config path`.
+:::
+
+### Run a Private Package
+
+Now, run a private package! Get the locator of the private package you have access to that you'd like to run that. For this example, we'll assume there's a private package at `github.com/tedim52/my-private-package`
+```
```
Comment (h4ck3rk3y) [inline docs/docs/guides/running-private-packages.md:52]: maybe replace tedim52 with "private-author" or something generic

File: docs/docs/guides/running-private-packages.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+![github-one-time-code](../../static/img/guides/one-time-code.jpg)
+
+A GitHub screen should pop up in your browser instructing you to enter the one-time code you just copied in your terminal. 
+
+![github-enter-code](../../static/img/guides/github-enter-code.jpg)
+
+After entering the code, GitHub will prompt you to authorize Kurtosis CLI. Kurtosis CLI only requests read access to public and private repositories. 
+
+:::info Private Packages within GitHub Orgs
+If the private package you'd like to develop on lives in a GitHub org, the organization must also authorize Kurtosis CLI.
```
Comment (h4ck3rk3y) [inline docs/docs/guides/running-private-packages.md]: How do we do this?

**Review by h4ck3rk3y — APPROVED** [review]

File: cli/cli/helpers/oauth/oauth.go (hunk 1 above)
Comment (tedim52) [reply cli/cli/helpers/oauth/oauth.go:43]: No `repo:read` scope exists :( a la: https://kurtosistech.slack.com/archives/D040G0YT5G8/p1708451376845819)
https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/scopes-for-oauth-apps

File: cli/cli/helpers/oauth/oauth.go (hunk 2 above)
Comment (tedim52) [reply cli/cli/helpers/oauth/oauth.go:54]: Yes, but we want the user to copy the one time code first and read the instruction before entering, so they're not confused by what to do next: https://github.com/kurtosis-tech/kurtosis/issues/2181

File: cli/cli/commands/github/login/login.go (hunk 3 above)
Comment (tedim52) [reply cli/cli/commands/github/login/login.go:73]: ack

File: docs/docs/cli-reference/service-logs.md (hunk 5 above)
Comment (tedim52) [reply docs/docs/cli-reference/service-logs.md:19]: ack

File: docs/docs/guides/running-private-packages.md (hunk 8 above)
Comment (tedim52) [reply docs/docs/guides/running-private-packages.md]: added info

**Pull request merged**